### PR TITLE
Fix + unit tests for parsing_and_checking_json

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [ ] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Unit tests

## Description

- Fix for `parsing_and_checking_json` function: `TypeError` is not caught
- Added unit tests with 100% coverage for this function

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Run `make test-unit` as usual

### CODE REVIEW

File: tests/unit/utils/test_json_tools.py [hunk 1]
```diff
@@ -37,3 +38,53 @@ def test_parse_json_markdown_negative_test_cases(json):
...
+    ('```{"x":42, "y":""}```', {"x": 42, "y": ""}, ["x", "y"]),
+    ('\n{"x":42, "y":""}\n', {"x": 42, "y": ""}, ["x", "y"]),
+    ('\t{"x":42, "y":""}\t', {"x": 42, "y": ""}, ["x", "y"]),
+    ('```json{"x":42, "y":""}```', {"x": 42, "y": ""}, ["x", "y"]),
+)
+
+
+@pytest.mark.parametrize("json,expected,keys", correct_inputs_with_keys)
+def test_parse_and_check_json_markdown_check_keys(json, expected, keys):
+    """Check parsing incorrect input and checking if resulting object contains given keys."""
```
Comment (bparees) [inline tests/unit/utils/test_json_tools.py:66]:
```suggestion
    """Check parsing correct input and checking if resulting object contains given keys."""
```

?

Comment (tisnik) [reply tests/unit/utils/test_json_tools.py]: nice catch! Fixed

**Pull request merged**